data_steward/tools/create_tier.py

```diff
@@ -0,0 +1,130 @@
+        )
+        raise argparse.ArgumentTypeError
+    return arg_value
+
+
+def create_tier(credentials_filepath, project_id, tier, input_dataset,
+                release_tag, deid_stage):
+    """
+    This function is hte main entry point for the deid process.
+    It passes the required parameters to the implementing functions.
+
+    :param credentials_filepath: filepath to credentials to access GCP
+    :param project_id: project_id associated with the input dataset
+    :param tier: controlled or registered tier intended for the output dataset
+    :param input_dataset: name of the input dataset
+    :param release_tag: release tag for dataset in the format of YYYYq#r#
+    :param deid_stage: deid stage (deid, base or clean)
+    :return: name of created controlled or registered dataset
+    """
+
+    # validation of params
+    validate_tier_param(tier)
+    validate_deid_stage_param(deid_stage)
+
+
+def parse_deid_args(args=None):
+    parser = argparse.ArgumentParser(
+        description='Parse deid command line arguments')
+    parser.add_argument('-c',
+                        '--credentials_filepath',
+                        dest='credentials_filepath',
+                        action='store',
+                        help='file path to credentials for GCP to access BQ',
+                        required=True)
+    parser.add_argument('-p',
+                        '--project_id',
+                        action='store',
+                        dest='project_id',
+                        help=('Project associated with the '
+                              'input dataset.'),
+                        required=True)
+    parser.add_argument('-t',
+                        '--tier',
+                        action='store',
+                        dest='tier',
+                        help='controlled or registered tier',
+                        required=True,
+                        choices=['controlled', 'registered'])
+    parser.add_argument('-i',
+                        '--idataset',
+                        action='store',
+                        dest='idataset',
+                        help='Name of the input dataset',
+                        required=True)
+    parser.add_argument(
+        '-r',
+        '--release_tag',
+        action='store',
+        dest='release_tag',
+        help='release tag for dataset in the format of YYYYq#r#',
+        required=True,
+        type=valid_release_tag)
+    parser.add_argument('-d',
+                        '--deid_stage',
+                        action='store',
+                        dest='deid_stage',
+                        help='deid stage (deid, base or clean)',
+                        required=True,
+                        choices=['deid', 'base', 'clean'])
+    return parser.parse_args(args)
+
+
+def main(raw_args=None):
+    args = parse_deid_args(raw_args)
+    create_tier(args.credentials_filepath, args.project_id, args.tier,
```

Comment:
Probably add the piece about configuring the pipeline here, after parsing the arguments.  To make it dynamic, you may want to add an argument allowing the logs to print to the console, and pass that to the configure function.

Reply:
Let me know if I did this correctly.